java/src/main/java/com/google/oak/crypto/SenderEncryptor.java

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.google.oak.crypto;
+
+import com.google.oak.util.Result;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+/**
+ * Encryptor class for encrypting sender requests that will be sent to the recipient and decrypting
+ * recipient responses that are received by the sender. Each Encryptor object corresponds to a
+ * single crypto session between the sender and the recipient.
+ *
+ * Sequence numbers for requests and responses are incremented separately, meaning that there could
+ * be multiple responses per request and multiple requests per response.
+ */
+public class SenderEncryptor {
+  // Info string used by Hybrid Public Key Encryption.
+  private static final String OAK_HPKE_INFO = "Oak Hybrid Public Key Encryption v1";
+
+  private final byte[] serialized_recipient_public_key;
+
+  /**
+   * Creates a new instance of [`SenderEncryptor`].
+   *
+   * @param serialized_recipient_public_key a NIST P-256 SEC1 encoded point public key.
+   */
+  public SenderEncryptor(final byte[] serialized_recipient_public_key) {
+    this.serialized_recipient_public_key = serialized_recipient_public_key;
+  }
+
+  /**
+   * Encrypts `plaintext` and authenticates `associated_data` using AEAD.
+   * <https://datatracker.ietf.org/doc/html/rfc5116>
+   *
+   * @param plaintext the input byte array to be encrypted
+   * @param associated_data the input byte array with associated data to be authenticated
+   * @return a serialized {@code EncryptedRequest} message wrapped in a {@code Result}
+   * TODO(#3843): Return unserialized proto messages once we have Java encryption without JNI.
```

Comment:
I suggest leaving the TODO out of the doc comment. I'd move this below after the other TODO.